src/main/java/com/internet/shop/dao/jdbs/ShoppingCartJdbcImpl.java

```diff
@@ -0,0 +1,208 @@
+package com.internet.shop.dao.jdbs;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartJdbcImpl implements ShoppingCartDao {
+    @Override
+    public boolean delete(ShoppingCart cart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE shopping_carts SET deleted = TRUE WHERE cart_id = ? "
+                            + "AND deleted = FALSE");
+            statement.setLong(1, cart.getId());
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting order by id: "
+                    + cart.getId() + " was failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteProducts(id);
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE shopping_carts SET deleted = TRUE WHERE cart_id = ? "
+                            + "AND deleted = FALSE");
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting order by id: "
+                    + id + " was failed", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        Optional<ShoppingCart> cart = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM shopping_carts WHERE user_id = ? AND deleted = FALSE");
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                cart = Optional.of(getCartFromResult(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Finding cart by user id: "
+                    + userId + " was failed", e);
+        }
+        if (cart.isEmpty()) {
+            return cart;
+        }
+        cart.get().setProducts(getProducts(cart.get().getId()));
+        return cart;
+    }
```

Comment:
це можна трішки спростити:
```suggestion
    public Optional<ShoppingCart> getByUserId(Long userId) {
        ShoppingCart cart = null;
        try (Connection connection = ConnectionUtil.getConnection()) {
            PreparedStatement statement = connection.prepareStatement(
                    "SELECT * FROM shopping_carts WHERE user_id = ? AND deleted = FALSE");
            statement.setLong(1, userId);
            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next()) {
                cart = getCartFromResult(resultSet);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Finding cart by user id: "
                    + userId + " was failed", e);
        }
        if (cart != null) {
            cart.setProducts(getProducts(cart.getId()));
        }
        return Optional.ofNullable(cart);
    }
```

Reply:
done